src/Raven.Server/Raven.Server.csproj

```diff
@@ -240,4 +240,7 @@
     <ConcurrentGarbageCollection>false</ConcurrentGarbageCollection>
     <RetainVMGarbageCollection>false</RetainVMGarbageCollection>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(AllowEncryptedOverHttp)' == 'True'">
+    <DefineConstants>$(DefineConstants);ALLOW_ENCRYPTED_OVER_HTTP</DefineConstants>
+  </PropertyGroup>
```

Comment:
You could actually generalize this whole PR into a few lines of csproj, no PS code modification whatsoever, by using the env var directly in here. 

See: https://learn.microsoft.com/en-us/visualstudio/msbuild/how-to-use-environment-variables-in-a-build?view=vs-2022

This is good because:
- you don't modify powershell pipeline and drop all the above changes
- you can easily add other constants, mix them etc.

However I think you'd rather after modifying *Directory.build.props* than just the server then.

Consider the following program:
```
#if (OPTION1)

Console.WriteLine("Option 1");

#endif

#if (OPTION2)

Console.WriteLine("Option 2");

#endif
```

and its project file
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

 <PropertyGroup Condition="'$(RAVEN_BuildOptions)' != ''">
    <DefineConstants>$(DefineConstants);$(RAVEN_BuildOptions)</DefineConstants>
  </PropertyGroup>

</Project>
```

And then this output:
```
C:\work\debug\buildopt> $env:RAVEN_BuildOptions='OPTION1'
C:\work\debug\buildopt> dotnet run
Option 1
C:\work\debug\buildopt> $env:RAVEN_BuildOptions='OPTION1,OPTION2'
C:\work\debug\buildopt> dotnet run
Option 1
Option 2
C:\work\debug\buildopt>
```

Ain't that cool?